compiler/src/dotty/tools/dotc/ast/Desugar.scala

```diff
@@ -925,6 +936,22 @@ object desugar {
     }
   }
 
+  /** The selector of a match, which depends of the given `checkMode`.
```

Comment:
The fact that `@unchecked` is also added to irrefutable patterns really confused me until I understood that annotating the selector only affects exhaustivity checks and not irrefutability checks. Perhaps we could note this in the doc?

```suggestion
  /** The selector of a match, which depends of the given `checkMode`.
   *
   * The @unchecked annotation is added whenever `checkMode` is not `Exhaustive` to silence
   * unnecessary inexhaustive match warnings.
```
```